### PR TITLE
Feat replication basic metrics

### PR DESCRIPTION
**Pull request by byo**

### CODE REVIEW

File: pkg/replication/metrics.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
+package replication
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// TODO: Metrics should be put behind abstract metrics interfaces to avoid dependency on
+// prometheus inside embedded package
```
Comment (jeroiraz) [inline pkg/replication/metrics.go:9]: in this case it's not the embedded package

**Review by jeroiraz — APPROVED** [review]
LGTM

File: pkg/database/database.go [hunk 2]
```diff
@@ -123,7 +123,7 @@ type DB interface {
 	WaitForIndexingUpto(txID uint64, cancellation <-chan struct{}) error
 
 	TxByID(req *schema.TxRequest) (*schema.Tx, error)
-	ExportTxByID(req *schema.ExportTxRequest) (txbs []byte, mayCommitUpToTxID uint64, mayCommitUpToAlh [sha256.Size]byte, err error)
+	ExportTxByID(req *schema.ExportTxRequest) (txbs []byte, mayCommitUpToTxID uint64, mayCommitUpToAlh [sha256.Size]byte, committedTxID uint64, err error)
```
[the remark this answers is not in the document]
Comment (byo) [reply pkg/database/database.go]: Updated

File: pkg/replication/metrics.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+		primaryCommittedTxID:     _metricsReplicationPrimaryCommittedTxID.WithLabelValues(dbName),
+		allowCommitUpToTxID:      _metricsAllowCommitUpToTxID.WithLabelValues(dbName),
+	}
+}
+
+// reset ensures all necessary gauges are zeroed
+func (m *metrics) reset() {
+	m.replicators.Set(0)
+	m.replicatorsActive.Set(0)
+	m.replicatorsInRetryDelay.Set(0)
```
Comment (jeroiraz) [inline pkg/replication/metrics.go:99]: primaryCommittedTxID and allowCommitUpToTxID do not need to be reset?

File: pkg/replication/replicator.go [hunk 4]
```diff
@@ -319,15 +359,27 @@ func (txr *TxReplicator) fetchNextTx() error {
 	if syncReplicationEnabled {
 		md := exportTxStream.Trailer()
 
-		if len(md.Get("may-commit-up-to-txid-bin")) == 0 || len(md.Get("may-commit-up-to-alh-bin")) == 0 {
-			return fmt.Errorf("master is not running with synchronous replication")
+		if len(md.Get("may-commit-up-to-txid-bin")) == 0 ||
+			len(md.Get("may-commit-up-to-alh-bin")) == 0 ||
+			len(md.Get("committed-txid-bin")) == 0 {
```
Comment (jeroiraz) [inline pkg/replication/replicator.go:364]: is `committed-txid-bin` set?

File: pkg/server/stream_replication.go [hunk 5]
```diff
@@ -38,12 +38,19 @@ func (s *ImmuServer) ExportTx(req *schema.ExportTxRequest, txsServer schema.Immu
...
 	defer func() {
 		if req.FollowerState != nil {
-			var bTxID [8]byte
-			binary.BigEndian.PutUint64(bTxID[:], mayCommitUpToTxID)
+			var bMayCommitUpToTxID [8]byte
+			binary.BigEndian.PutUint64(bMayCommitUpToTxID[:], mayCommitUpToTxID)
+
+			var bCommittedTxID [8]byte
+			state, err := db.CurrentState()
+			if err != nil {
```
Comment (jeroiraz) [inline pkg/server/stream_replication.go:46]: `if err != nil {` would be `if err == nil {` ?

File: pkg/replication/metrics.go (hunk 3 above)
Comment (byo) [reply pkg/replication/metrics.go:99]: My assumption was false about the gauge - once it's instantiated for given db it starts appearing in prometheus output. I've added reset to 0 for missing metrics

**Review by jeroiraz — DISMISSED** [review]

**Pull request merged**